pkg/bloomgateway/bloomgateway.go

```diff
@@ -342,9 +396,9 @@ outer:
 		case res := <-resCh:
```

Comment:
In case of an error executing any of the subtasks, shouldn't we cancel the context so other subtasks stop as well?

Reply:
Cancellation of tasks does not really work yet, that's due to how we rely on getting results from the channel and not passing the context to the task.
Yes, it's an issue indeed, but I would make a separate PR to fix that properly. I thought it would exceed the scope of this PR.